transaction/std/src/memo/gift_code_funding.rs

```diff
@@ -30,36 +32,68 @@ impl RegisteredMemoType for GiftCodeFundingMemo {
 }
 
 impl GiftCodeFundingMemo {
-    /// Create a new gift funding code memo
-    pub fn new(tx_out_public_key: &RistrettoPublic, note: &str) -> Result<Self, MemoError> {
+    /// Number of bytes in the memo
+    pub const MEMO_DATA_LEN: usize = 64;
+
+    /// Number of bytes used to represent the gift code TxOut hash
+    pub const HASH_DATA_LEN: usize = 4;
+
+    /// Number of bytes used to represent the fee paid when funding the gift
+    /// code
+    pub const FEE_DATA_LEN: usize = 7;
+
+    /// Number of bytes used to represent the utf-8 note
+    pub const NOTE_DATA_LEN: usize = 53;
+
+    /// Byte offset of the note
+    pub const NOTE_OFFSET: usize = Self::HASH_DATA_LEN + Self::FEE_DATA_LEN;
+
+    /// The max fee (i.e. the maximum value a 56 bit number can contain)
+    pub const MAX_FEE: u64 = u64::MAX >> 8;
+
+    /// Create a new gift funding memo
+    pub fn new(
+        tx_out_public_key: &RistrettoPublic,
+        fee: u64,
+        note: &str,
+    ) -> Result<Self, MemoError> {
+        // Check if the fee we're setting is greater than the max fee
+        if fee > Self::MAX_FEE {
+            return Err(MemoError::MaxFeeExceeded(Self::MAX_FEE, fee));
+        }
         // Check if note is of valid length and initialize memo data
         if note.len() > Self::NOTE_DATA_LEN {
             return Err(MemoError::BadLength(note.len()));
         }
+
         let mut memo_data = [0u8; Self::MEMO_DATA_LEN];
-        // Compute TxOut hash and store it into the memo data
-        memo_data[0..Self::HASH_DATA_LEN]
+
+        // Compute TxOut hash and put it into the memo data
+        memo_data[..Self::HASH_DATA_LEN]
             .copy_from_slice(&tx_out_public_key_short_hash(tx_out_public_key));
 
+        // Put fee into memo
+        memo_data[Self::HASH_DATA_LEN..Self::NOTE_OFFSET]
+            .copy_from_slice(&fee.to_be_bytes()[1..(Self::FEE_DATA_LEN + 1)]);
```

Comment:
Optional/Equivalent: Use an inclusive range.
```suggestion
            .copy_from_slice(&fee.to_be_bytes()[1..=Self::FEE_DATA_LEN]);
```

Reply:
Done. I actually didn't know about this, thanks for the tip!